mantidproject/mantid
# Indirect OSIRIS Diffraction -  Allow for multiple can subtractions

OSIRIS Diffraction Reduction (diffonly mode) now supports loading multiple container files. The tests have also been updated for this change as well as to cover using multiple vanadium and samples (which was available before but untested in the unit test)

**To test:**
- Running the following script should test the algorithm

```
wks = OSIRISDiffractionReduction(Sample=['OSI10203.raw','OSI10204.raw'],
                                                       CalFile='osiris_041_RES10.cal',
                                                       Vanadium=['OSI10156.raw','OSI10157.raw'],
                                                       Container=['OSI10241.raw','OSI10242.raw'],
                                                       ContainerScaleFactor=0.5,
                                                       SpectraMin=3,
                                                       SpectraMax=361,
                                                       DetectDRange=True)
```
- The interface (Interfaces > Indirect > Diffraction) also needs to be checked.
  - Change the Instrument to OSIRIS/diffraction/diffonly
  - Add in the Run Numbers(`10203-10204`)/Vanadium(`10156-10157`) and Container(`10241-10242`)
  - Change `Spectra Max` to 361
  - Add the Cal File (`osiris_041_RES10.cal`) available from `ExternalData/Testing/Data/SystemTest/`
  - If you are using the run/van and can files from the unit test data you may be required to untick `Load Log Files` option.
  - Click `run`
  - All should run fine without error

Fixes #15957.

[Release Notes](https://github.com/mantidproject/mantid/blob/15957_OSIRIS_Diff_multi_can_subraction/docs/source/release/v3.7.0/indirect_inelastic.rst#diffraction)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
